### ISSUE TITLE
Firefox on Windows complains that extension is corrupted

### ISSUE THREAD

**Issue by caleb** [gab-ai-inc/gab-dissenter-extension#75]
I'm using Firefox 66.0.3 on Windows 10, and when I try to install dissenter (dissenter_extension-0.1.9-fx) I get a popup telling me that it is corrupt. "This add-on cannot be installed because it appears to be corrupt"

I had dissenter on this browser, but I think Firefox disabled it.

I've downloaded it multiple times and restarted the browser fully, but it says the same thing.

**Comment by JDog2pt0** [comment]
I'm having this issue with Waterfox 56.2.9 as well. I've got 0.1.1 installed, but when I try to download the extension for either FF or WF, I get the "...appears to be corrupt" notice.

**Comment by Shawn-McKinley** [comment]
I think Firefox made a "security" upgrade, unbeknownst to me, in the past couple of days. NoScript and a couple of other extensions were also "removed for my safety".

Looks like it might finally be time to lose Firefox, even though I have championed it over Chrome for the life it that browser. Thanks Mozilla, for giving me the kick I needed to drop Firefox.

**Comment by noreasternhumidity** [comment]
I think I see what's going on here.  The ENTIRE Firefox add-on process has been modified to disallow any "unsigned" add-ons.  That would include anything that doesn't pass through their "app store", such as Dissenter.  In other words, I suspect that Firefox is blocking at browser level anything that Mozilla doesn't like.

**Comment by Ethel173** [comment]
> I think I see what's going on here. The ENTIRE Firefox add-on process has been modified to disallow any "unsigned" add-ons - and you are seeing some "collateral damage" as it spits up all over itself. 

seconded, got an email May 2, 2019, 11:56 AM

tldr of which is 

- changes will go into effect on June 10, 2019. (which is clearly a lie)
>     We will no longer accept extensions that contain obfuscated code. We will continue to allow minified, concatenated, or otherwise machine-generated code as long as the source code is included. Please see our updated source code submission page for more details.
>     Continued use of obfuscated code may result in your add-on being blocked. 
>     We will be blocking extensions more proactively if add-ons are found to be in violation of our policies. We will be casting a wider net, and will err on the side of user security when determining whether or not to block.

oh well time to migrate to waterfox i guess

**Comment by caleb** [comment]
If this is the case, it wouldn’t be so bad if they just said that this extension is unsigned, and gave me a button to install it anyway.

The removal from the add-on repository is pretty troubling to me for the future of Firefox. Maybe it’s time to look at alternatives.

Remember when Mozilla /was/ the alternative you went to when you wanted more flexibility and less corporate politics back in 2003?

**Comment by Berbecarul** [comment]
Just downloaded 66.0b9 dev edition of Mozilla from https://download-installer.cdn.mozilla.net/pub/devedition/releases/ and seems to be unaffected. Might not give up on Mozilla just yet. I wonder how the dissenter extension could meet their requirements.

**Comment by Ethel173** [comment]
seems that this was actually an error on behalf of mozilla [accidentally disabling all addons because a certificate expired](https://www.engadget.com/2019/05/03/firefox-extension-add-on-cert/)

restart firefox and you should be able to install the addon

**Comment by fmra357** [comment]
I've tried to side load the extension on three different versions of firefox (66.0.3, dev edition, and waterfox) and each time its telling me the install is corrupted. Tried it with the install button and by downloading it and dropping it on the browser, neither works. Is the extension broken or did FF hardcode the ban?

**Comment by thatcomicsguy** [comment]
I got it going.  I'm using the Firefox Nightly Build here:
https://www.mozilla.org/en-US/firefox/nightly/all/

1. Type in: "about:config" in the address bar.
2. Search for: "xpinstall.signatures.required"
3. Set that to "False".
4. Go to Tools/addons and use the gear dropdown menu and select "Debug Add-ons"
or alternatively, just type "about:debugging#/runtime/this-firefox" into the search bar.
5. Check the option, "Enable Extension Debugging"
6. Click, "Load Temporary Add-on" and select the dissenter xpi:  "dissenter_extension-0.1.5-fx.xpi"
 
And then you're up and running.

For now, anyway.  I'm sure the Mozilla critters will find a way to crush free speech yet.

**Comment by Davidj361** [comment]
> I got it going. It will work with the regular current Firefox (version 66.0.3)
> 
>     1. Type in: "about:config" in the address bar.
> 
>     2. Search for: "xpinstall.signatures.required"
> 
>     3. Set that to "False".
> 
>     4. Go to Tools/addons and use the gear dropdown menu and select "Debug Add-ons"
> 
>     5. Check the option, "Enable Extension Debugging"
> 
>     6. Click, "Load Temporary Add-on" and select the dissenter xpi:  "dissenter_extension-0.1.5-fx.xpi"
> 
> 
> And then you're up and running. (Incidentally, following this same process allows you to get all your other add-ons working again while Mozilla is running around putting out fires.)
> 
> If you want to install your old add-ons, you need to download the .xpi for it (the add-on file). You can get those simply by right clicking on the "+Add To Firefox" button and selecting "Save Link As".
> 
> The only limitation with this fix is that if you close and re-start Firefox, you'll need to re-enable extension debugging. That's annoying, but I tend to leave my browser running for days at a time, so it's just a matter of keeping the .xpi files somewhere handy.
> 
> Anyway, this all works for now. Though I'm sure the Mozilla critters will find a way to crush free speech yet.

This should be on the main website by now. I'm even having this issue with IceCat 60.3.0esr (64-bit)

**Comment by Davidj361** [comment]
Getting the same issue on IceCat 60.3.0esr (64-bit) when trying to install the addon.
